subworkflows/local/circrna_discovery.nf

```diff
@@ -0,0 +1,227 @@
+include { STAR_ALIGN as STAR_1ST_PASS      } from '../../modules/nf-core/star/align/main'
+include { STAR_ALIGN as STAR_2ND_PASS      } from '../../modules/nf-core/star/align/main'
+include { SJDB as STAR_SJDB                } from '../../modules/local/star/sjdb/main'
+include { STAR_ALIGN as DCC_1ST_PASS       } from '../../modules/nf-core/star/align/main'
+include { STAR_ALIGN as DCC_2ND_PASS       } from '../../modules/nf-core/star/align/main'
+include { SJDB as DCC_SJDB                 } from '../../modules/local/star/sjdb/main'
+include { STAR_ALIGN as DCC_MATE1_1ST_PASS } from '../../modules/nf-core/star/align/main'
+include { STAR_ALIGN as DCC_MATE1_2ND_PASS } from '../../modules/nf-core/star/align/main'
+include { SJDB as DCC_MATE1_SJDB           } from '../../modules/local/star/sjdb/main'
+include { STAR_ALIGN as DCC_MATE2_1ST_PASS } from '../../modules/nf-core/star/align/main'
+include { STAR_ALIGN as DCC_MATE2_2ND_PASS } from '../../modules/nf-core/star/align/main'
+include { SJDB as DCC_MATE2_SJDB           } from '../../modules/local/star/sjdb/main'
+include { DCC                              } from '../../modules/local/dcc/dcc/main'
+include { DCC_FILTER                       } from '../../modules/local/dcc/filter/main'
+include { CIRCEXPLORER2_REFERENCE as MAPSPLICE_REFERENCE } from '../../modules/local/circexplorer2/reference/main'
+include { MAPSPLICE_ALIGN                  } from '../../modules/local/mapsplice/align/main'
+include { CIRCEXPLORER2_PARSE as MAPSPLICE_PARSE } from '../../modules/nf-core/circexplorer2/parse/main'
+include { CIRCEXPLORER2_ANNOTATE as MAPSPLICE_ANNOTATE } from '../../modules/nf-core/circexplorer2/annotate/main'
+include { CIRCEXPLORER2_FILTER as MAPSPLICE_FILTER } from '../../modules/local/circexplorer2/filter/main'
+include { FASTA                            } from '../../modules/local/fasta/main'
+include { MERGE_TOOLS                      } from '../../modules/local/count_matrix/merge_tools/main'
+include { COUNTS_COMBINED                  } from '../../modules/local/count_matrix/combined/main'
+include { COUNTS_SINGLE                    } from '../../modules/local/count_matrix/single/main'
+
+workflow CIRCRNA_DISCOVERY {
+
+    take:
+    reads
+    fasta
+    gtf
+    bowtie_index
+    bowtie2_index
+    bwa_index
+    chromosomes
+    hisat2_index
+    segemehl_index
+    star_index
+    bsj_reads
+    tool_filter
+    duplicates_fun
+    exon_boundary
+
+    main:
+    ch_versions = Channel.empty()
+
+    //
+    // SEGEMEHL WORKFLOW:
+    //
+    SEGEMEHL_ALIGN( reads, fasta, segemehl_index )
+    segemehl_filter = SEGEMEHL_ALIGN.out.results.map{ meta, results ->  meta.tool = "segemehl"; return [ meta, results ] }
+    SEGEMEHL_FILTER( segemehl_filter, bsj_reads )
+
+    ch_versions = ch_versions.mix(SEGEMEHL_ALIGN.out.versions)
+
+    //
+    // STAR WORFKLOW:
+    //
+
+    STAR_1ST_PASS( reads, star_index, gtf, true, '', '' )
```

Comment:
the empty and "true" statements are a bit hard to understand. How about assigning variables to them

Reply:
nice idea